idea-plugin/src/main/kotlin/arrow/meta/dsl/ide/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -12,7 +12,24 @@ import com.intellij.openapi.progress.ProgressManager
 import com.intellij.psi.PsiElement
 import javax.swing.Icon
 
+data class LineMarker(val icon: Icon,
+                      val message: String,
+                      val placed: GutterIconRenderer.Alignment = GutterIconRenderer.Alignment.RIGHT)
+
 interface LineMarkerSyntax {
+
+  fun IdeMetaPlugin.lineMarker(
```

Comment:
We could make this signature better typed using a signature like `filterMap` instead of a predicate. It'll require a new `extensionProvider`. impl tho.

```
fun <A> IdeMetaPlugin.lineMarker(
     mapFilter: (psi: PsiElement) -> A?,
     marker: (a: A) -> LineMarker
): ExtensionPhase
```

Reply:
I can figure  something out to have the type A reified

Reply:
@nomisRev I hope this API is friendlier